internal/controllers/utils/utils.go

```diff
@@ -1155,3 +1156,43 @@ func SetCloudManagerGenerationStatus(ctx context.Context, c client.Client, nodeP
 	}
 	return nil
 }
+
+// ParseJSON parse the yaml in arguments to a data structure T
+func ExtractSubSchema(mainSchema []byte, node string) (subSchema []byte, err error) {
```

Comment:
This should be updated to work with the valid schema (keys defined inside of "properties"), and could we change `node` to subSchemaKey?

Reply:
Fixed